npm/private/npm_translate_lock_state.bzl

```diff
@@ -155,7 +155,9 @@ def _init_pnpm_labels(label_store, rctx):
     #  Label("@nodejs_host//:bin/node")
     label_store.add("host_node", Label("@{}_{}//:bin/node".format(rctx.attr.node_toolchain_prefix, repo_utils.platform(rctx))))
 
-    label_store.add("pnpm_entry", Label("@pnpm//:package/bin/pnpm.cjs"))
+    pnpm_repo = rctx.attr.use_pnpm if rctx.attr.use_pnpm else Label("@pnpm")
+
+    label_store.add("pnpm_entry", pnpm_repo.relative(":package/bin/pnpm.cjs"))
```

Comment:
This `":package/bin/pnpm.cjs"` is what I think still needs discussion I think.

Putting the path here means the public API only needs to specify `@pnpm-repo-name`, but then the public API can't change the path at all. I think `@pnpm//:pnpm` (the default label) pointing to the bin would be great, so the default is nice but can still be customized, but currently that default label is [already defined](https://github.com/aspect-build/rules_js/blob/93e53bbb60d966e6b4b9d3b0440739009b62bf5f/npm/private/pnpm_repository.bzl#L40) as something else. I'm trying that out atm, but other ideas/feedback would be great...

Reply:
Since labels can not be evaluated in repo rules I think we either need to either hardcode the `:package/bin/pnpm.cjs"` here when adding it to the label or when the public API is used.

See the diff swapping between my 2 ideas: https://github.com/aspect-build/rules_js/compare/ae7a283ee5285380c3ce0fc749508818ce547ec2..db67dda7077b201ff5323fa040193fac0aa2cfd7